lib/src/annotate.rs

```diff
@@ -116,10 +120,38 @@ pub fn get_annotation_for_file(
     domain: &Rc<ResolvedRevsetExpression>,
     file_path: &RepoPath,
 ) -> Result<FileAnnotation, RevsetEvaluationError> {
-    let mut source = Source::load(starting_commit, file_path)?;
+    let source = Source::load(starting_commit, file_path)?;
+    compute_file_annotation(repo, starting_commit.id(), domain, file_path, source)
+}
+
+/// Get line by line annotations for a specific file path starting with the
+/// given content.
+///
+/// The file content at the `starting_commit` is set to `starting_text`. This is
+/// typically one of the file contents in the conflict or merged-parent tree.
+///
+/// See [`get_annotation_for_file()`] for the other arguments.
```

Comment:
Out of scope for this PR, but perhaps an alternative is to add an option to continue annotating older version, so we'd record information about every commit where a line was modified, kind of like a weave. That would also be useful for a `jj annotate --interactive/--tui` that lets you view the annotated file at a previous commit.

Reply:
Yes. Even with the current implementation, annotation process can be resumed if the result contained remainder of `CommitSourceMap`, but I'm not sure if that's sufficient for interactive use case or weave-like merging.